packages/pigeon/test/pigeon_lib_test.dart

```diff
@@ -416,4 +416,26 @@ void main() {
     expect(parseResult.errors[0].message, contains('Generic fields'));
     expect(parseResult.errors[0].lineNumber, isNotNull);
   });
+
+  test('error with static field', () {
```

Comment:
It might make sense to transition these tests to look more like this one.  I tried to investigate creating the files in memory to avoid writing to disk but it didn't work out.  I might try giving it one more look to make sure.  (I think it's a problem that the analyzer wants a path instead of a File).